src/openrct2-ui/windows/Map.cpp

```diff
@@ -1188,89 +1188,92 @@ static void window_map_set_land_rights_tool_update(int32_t x, int32_t y)
  *
  *  rct2: 0x00666EEF
  */
-static void place_park_entrance_get_map_position(
-    int32_t x, int32_t y, int16_t* mapX, int16_t* mapY, int16_t* mapZ, int32_t* direction)
+static CoordsXYZD place_park_entrance_get_map_position(ScreenCoordsXY screenCoords)
 {
-    sub_68A15E(x, y, mapX, mapY);
-    if (*mapX == LOCATION_NULL)
-        return;
-
-    auto surfaceElement = map_get_surface_element_at(*mapX >> 5, *mapY >> 5);
-    *mapZ = surfaceElement->GetWaterHeight();
-    if (*mapZ == 0)
+    CoordsXYZD parkEntranceMapPosition{ 0, 0, 0, INVALID_DIRECTION };
+    const CoordsXY mapCoords = sub_68A15E(screenCoords);
+    parkEntranceMapPosition = { mapCoords.x, mapCoords.y, 0, INVALID_DIRECTION };
+    if (parkEntranceMapPosition.x == LOCATION_NULL)
+        return parkEntranceMapPosition;
+
+    auto surfaceElement = map_get_surface_element_at(mapCoords);
+    parkEntranceMapPosition.z = surfaceElement->GetWaterHeight() * 8;
+    if (parkEntranceMapPosition.z == 0)
     {
-        *mapZ = surfaceElement->base_height / 2;
+        parkEntranceMapPosition.z = surfaceElement->base_height * 8;
         if ((surfaceElement->GetSlope() & TILE_ELEMENT_SLOPE_ALL_CORNERS_UP) != 0)
         {
-            (*mapZ)++;
+            parkEntranceMapPosition.z++;
             if (surfaceElement->GetSlope() & TILE_ELEMENT_SLOPE_DOUBLE_HEIGHT)
             {
-                (*mapZ)++;
+                parkEntranceMapPosition.z++;
```

Comment:
```suggestion
                parkEntranceMapPosition.z += 16;
```

Reply:
Just to check: normally you would use a factor of 8 when converting from small to big coordinates.

Was parkEntranceMapPosition.z using half-tile units before?